src/js_subscriptions.hpp

```diff
@@ -0,0 +1,815 @@
+    static void get_name(ContextType, ObjectType, ReturnValue&);
+    static void get_object_type(ContextType, ObjectType, ReturnValue&);
+    static void get_query_string(ContextType, ObjectType, ReturnValue&);
+
+    PropertyMap<T> const properties = {
+        {"id", {wrap<get_id>, nullptr}},
+        {"createdAt", {wrap<get_created_at>, nullptr}},
+        {"updatedAt", {wrap<get_updated_at>, nullptr}},
+        {"name", {wrap<get_name>, nullptr}},
+        {"objectType", {wrap<get_object_type>, nullptr}},
+        {"queryString", {wrap<get_query_string>, nullptr}},
+    };
+};
+
+template <typename T>
+typename T::Object SubscriptionClass<T>::create_instance(ContextType ctx, realm::sync::Subscription subscription)
+{
+    return create_object<T, SubscriptionClass<T>>(ctx, new Subscription<T>(std::move(subscription)));
+}
+
+/**
+ * @brief Get the ID of the subscription
+ *
+ * @param ctx JS context
+ * @param object \ref ObjectType wrapping the Subscription
+ * @param return_value \ref ReturnValue wrapping a ObjectId containing the ID
+ */
+template <typename T>
+void SubscriptionClass<T>::get_id(ContextType ctx, ObjectType this_object, ReturnValue& return_value)
+{
+    auto sub = get_internal<T, SubscriptionClass<T>>(ctx, this_object);
+    return_value.set(sub->id());
+}
+
+/**
+ * @brief Get the created date of the subscription
+ *
+ * @param ctx JS context
+ * @param object \ref ObjectType wrapping the Subscription
+ * @param return_value \ref ReturnValue wrapping a Date containing the created date
+ */
+template <typename T>
+void SubscriptionClass<T>::get_created_at(ContextType ctx, ObjectType this_object, ReturnValue& return_value)
+{
+    auto sub = get_internal<T, SubscriptionClass<T>>(ctx, this_object);
+    return_value.set(Object::create_date(ctx, sub->created_at().get_nanoseconds()));
+}
+
+/**
+ * @brief Get the updated date of the subscription
+ *
+ * @param ctx JS context
+ * @param object \ref ObjectType wrapping the Subscription
+ * @param return_value \ref ReturnValue wrapping a Date containing the updated date
+ */
+template <typename T>
+void SubscriptionClass<T>::get_updated_at(ContextType ctx, ObjectType this_object, ReturnValue& return_value)
+{
+    auto sub = get_internal<T, SubscriptionClass<T>>(ctx, this_object);
+    return_value.set(Object::create_date(ctx, sub->updated_at().get_nanoseconds()));
+}
+
+/**
+ * @brief Get the name of the subscription
+ *
+ * @param ctx JS context
+ * @param object \ref ObjectType wrapping the Subscription
+ * @param return_value \ref ReturnValue wrapping a string containing the name, or null if the name is not set
+ */
+template <typename T>
+void SubscriptionClass<T>::get_name(ContextType ctx, ObjectType this_object, ReturnValue& return_value)
+{
+    auto sub = get_internal<T, SubscriptionClass<T>>(ctx, this_object);
+    auto name = sub->name();
+
+    if (name == "") {
+        return_value.set_null();
+    }
+    else {
+        return_value.set(std::string{name});
```

Comment:
Is the conversion to `std::string` needed?

Reply:
It seems so, without it I get lots of variations on `note: candidate function not viable: no known conversion from 'std::string_view' to 'const std::string'` – is there a better way convert a `std::string_view`?